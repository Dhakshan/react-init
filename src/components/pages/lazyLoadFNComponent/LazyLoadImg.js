import React, { Component } from 'react'
import PropTypes from 'prop-types'

// https://www.npmjs.com/package/react-lazy-load-image-component
import { LazyLoadImage } from 'react-lazy-load-image-component';

// const LazyLoadImg = ({src,alt,width,height,wrapperClassName,placeholderSrc,effect,afterLoad,beforeLoad,delayMethod,delayTime,threshold,useIntersectionObserver,placeholder,visibleByDefault}) => {
const LazyLoadImg = ({config}) => {
    return (
        <React.Fragment>
            <LazyLoadImage {...config}
            />
            {/* <LazyLoadImage 
                src={src} 
                alt={alt} 
                {...(threshold?{'threshold':threshold}:'')}   
                {...(width?{'width':width}:'')}  
                {...(height?{'height':height}:'')}   
                {...(wrapperClassName?{'wrapperClassName':wrapperClassName}:'')}   
                {...(placeholderSrc?{'placeholderSrc':placeholderSrc}:'')}   
                {...(effect?{'effect':effect}:'')}   
                {...(afterLoad?{'afterLoad':afterLoad}:'')}   
                {...(beforeLoad?{'beforeLoad':beforeLoad}:'')}   
                {...(delayMethod?{'delayMethod':delayMethod}:'')}   
                {...(delayTime?{'delayTime':delayTime}:'')}   
                {...(useIntersectionObserver?{'useIntersectionObserver':useIntersectionObserver}:'')}   
                {...(visibleByDefault?{'visibleByDefault':visibleByDefault}:'')}   
            /> */}
        </React.Fragment>
    );
};
LazyLoadImg.propTypes = {
    config : PropTypes.any
};
// LazyLoadImg.propTypes = {
//     src:PropTypes.string,
//     alt:PropTypes.string,
//     width:PropTypes.string,
//     height:PropTypes.string,
//     wrapperClassName:PropTypes.string,
//     placeholderSrc:PropTypes.string,
//     effect:PropTypes.string,
//     afterLoad:PropTypes.func,
//     beforeLoad:PropTypes.func,
//     delayMethod:PropTypes.func,
//     delayTime:PropTypes.number,
//     threshold:PropTypes.number,
//     useIntersectionObserver:PropTypes.bool,
//     visibleByDefault:PropTypes.bool
// };
export default LazyLoadImg;