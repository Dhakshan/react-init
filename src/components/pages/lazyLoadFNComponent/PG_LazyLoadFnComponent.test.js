import React from 'react';
import { shallow, mount } from 'enzyme';
import LazyLoadImg from './LazyLoadImg';

describe('<LazyLoadImg/>', () => {
    test('should render the component',()=>{
        const  component = shallow(<LazyLoadImg/>);
        expect(component).toMatchSnapshot();
    });
    test('props src should not empty',()=>{
        const config = {src:'Asset/images/some.png'};
        const  component = mount(<LazyLoadImg config={config}/>);
        expect(component.prop('config').src).toBeTruthy();
        component.unmount();
    });
    test('should show alt text when no config.src', () => {
        const config = {alt:'Product 1'};
        const component = mount(<LazyLoadImg config={config}/>);
        expect(component.prop('config').src).toBeFalsy();
        expect(component.prop('config').alt).not.toEqual('');
        expect(component.prop('config').alt).toEqual('Product 1');
        component.unmount();
    });
    test('config.threshold should be (0-1)', () => {
        const config = {threshold:0};
        const component = mount(<LazyLoadImg config={config}/>);
        expect(component.prop('config').threshold).not.toBeUndefined();
        expect(component.prop('config').threshold).not.toString();
        expect(component.prop('config').threshold).toBeGreaterThanOrEqual(0);
        expect(component.prop('config').threshold).toBeLessThanOrEqual(1);
        component.unmount();
    });
    // test('props config.delayMethod should be function', () => {
    //     const config = {delayMethod:()=>{}};
    //     const component = mount(<LazyLoadImg config={config}/>);
    //     expect(component.prop('config').delayMethod).not.toBeUndefined();
    //     expect(typeof component.prop('config').delayMethod).toBe('function');
    //     component.unmount();
    // });
    // test('props config.beforeload should be function', () => {
    //     const config = {beforeLoad:()=>{}};
    //     const component = mount(<LazyLoadImg config={config}/>);
    //     expect(component.prop('config').beforeLoad).not.toBeUndefined();
    //     expect(typeof component.prop('config').beforeLoad).toBe('function');
    //     component.unmount();
    // });
    // test('props config.afterload should be function', () => {
    //     const config = {afterLoad:()=>{}};
    //     const component = mount(<LazyLoadImg config={config}/>);
    //     expect(component.prop('config').afterLoad).not.toBeUndefined();
    //     expect(typeof component.prop('config').afterLoad).toBe('function');
    //     component.unmount();
    // });
    // test('should show alt text when no config.src', () => {
    //     const config = {alt:'Product 1'};
    //     const component = mount(<LazyLoadImg config={config}/>);
    //     console.log(component);
    //     component.unmount();
    // });
});

