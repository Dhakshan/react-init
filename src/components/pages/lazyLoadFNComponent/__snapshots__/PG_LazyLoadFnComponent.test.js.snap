// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LazyLoadImg/> check LazyLoadImage plugin 1`] = `
<t
  afterLoad={[Function]}
  alt="Product 1"
  beforeLoad={[Function]}
  delayMethod="throttle"
  delayTime={300}
  effect=""
  placeholderSrc={null}
  threshold={100}
  useIntersectionObserver={true}
  visibleByDefault={false}
  wrapperClassName=""
/>
`;

exports[`<LazyLoadImg/> should render the component 1`] = `
<Fragment>
  <t
    afterLoad={[Function]}
    beforeLoad={[Function]}
    delayMethod="throttle"
    delayTime={300}
    effect=""
    placeholderSrc={null}
    threshold={100}
    useIntersectionObserver={true}
    visibleByDefault={false}
    wrapperClassName=""
  />
</Fragment>
`;

exports[`<LazyLoadImg/> should render with alt config option 1`] = `
<t
  afterLoad={[Function]}
  alt="Product 1"
  beforeLoad={[Function]}
  delayMethod="throttle"
  delayTime={300}
  effect=""
  placeholderSrc={null}
  threshold={100}
  useIntersectionObserver={true}
  visibleByDefault={false}
  wrapperClassName=""
/>
`;
